### PR TITLE
MPDX-8208 - Allow the goal to be edited

### PR DESCRIPTION
**Pull request by dr-bizz**
## Description
In this PR, I allow for the goal of the appeal creator to be manually edited. I had to change how the other fields were setting the goal and I also cleaned up the file a little.

## Checklist:

- [ ] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [ ] I have applied the appropriate labels. (_Add the label "On Staging" to get the branch automatically merged into staging._)
- [ ] I have requested a review from another person on the project

### CODE REVIEW

**Comment by dr-bizz** [comment]
> We probably need a test that the goal can be modified manually.
Yes! I was trying to remember what test I needed to add, but it split my mind.

File: src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx [hunk 1]
```diff
@@ -76,8 +76,11 @@ export const calculateGoal = (
   adminPercentage: number,
 ): number => {
   const adminPercent = 1 - adminPercentage / 100;
+  const totalGoal =
+    Math.round(((initialGoal + letterCost) / adminPercent) * 100) / 100;
 
-  return Math.round(((initialGoal + letterCost) / adminPercent) * 100) / 100;
+  // Convert to string and remove ".00" if present
+  return parseFloat(totalGoal.toFixed(2));
```
[the remark this answers is not in the document]
Comment (dr-bizz) [reply src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx]: I should have removed that comment.

I have fixed this. I'm using Math.round to only show the first 2 decimal places.

File: src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx [hunk 2]
```diff
@@ -389,217 +400,251 @@ const AddAppealForm: React.FC<AddAppealFormProps> = ({
...
-                    className={classes.input}
-                    value={calculateGoal(
-                      initialGoal,
-                      letterCost,
-                      adminPercentage,
-                    ).toFixed(2)}
-                  />
-                </Box>
+      }): ReactElement => {
+        useEffect(() => {
```
Comment (canac) [inline src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx:404]: I guess this works without errors, but I don't love that we're using hooks not at the top-level of the component. Can we accept a little bit of duplication and set the goal field value in the `onChange` handlers for the other 3 inputs?

File: src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx [hunk 3]
```diff
@@ -389,217 +400,251 @@ const AddAppealForm: React.FC<AddAppealFormProps> = ({
...
+                    <TextField
+                      data-testid="goalInput"
+                      name="goal"
+                      type="number"
+                      label={t('Goal')}
+                      variant="outlined"
+                      size="small"
+                      className={classes.input}
+                      value={goal}
+                      onChange={(value) => {
```
Comment (canac) [inline src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx:540]:
```suggestion
                      onChange={(event) => {
```

**Review by canac — CHANGES_REQUESTED** [review]
When I type in a goal amount and try to create the appeal, I'm getting a GraphQL error: `Variable $attributes of type AppealCreateInput! was provided invalid value for amount (Could not coerce value "12.43" to Float)`. I'm not sure why it's not getting parsed to a number like the other inputs are.

File: src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx (hunk 2 above)
Comment (dr-bizz) [reply src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx:404]: I think, technically, it is at the top of the Formik component. But I can remove it and duplicate the code.

**Comment by dr-bizz** [comment]
I have error with the value not being a number

**Review by canac — APPROVED** [review]
Looks like it works now!

**Pull request merged**